packages/gatsby-plugin-benchmark-reporting/gatsby-node.js

```diff
@@ -0,0 +1,185 @@
+const { performance } = require(`perf_hooks`)
+const { execSync } = require("child_process")
+const path = require("path")
+
+const nodeFetch = require(`node-fetch`)
+const uuidv4 = require(`uuid/v4`)
+
+const bootstrapTime = performance.now()
+
+const BENCHMARK_REPORTING_URL =
+  process.env.BENCHMARK_REPORTING_URL === "cli"
+    ? undefined
+    : process.env.BENCHMARK_REPORTING_URL
+
+// Track the last received `api` because not all events in this plugin will receive one
+let lastApi
+function reportInfo(...args) {
+  ;(lastApi ? lastApi.reporter : console).info(...args)
+}
+function reportError(...args) {
+  ;(lastApi ? lastApi.reporter : console).error(...args)
+}
+
+class BenchMeta {
+  constructor() {
+    this.flushing = undefined // Promise of flushing if that has started
+    this.flushed = false // Completed flushing?
+    this.localTime = new Date().toISOString()
+    this.timestamps = {
+      // TODO: we should also have access to node's timing data and see how long it took before bootstrapping this script
+      bootstrapTime, // Start of this file
+      instantiationTime: performance.now(), // Instantiation time of this class
+      benchmarkStart: 0, // Start of benchmark itself
+      preInit: 0, // Gatsby onPreInit life cycle
+      preBootstrap: 0, // Gatsby onPreBootstrap life cycle
+      preBuild: 0, // Gatsby onPreBuild life cycle
+      postBuild: 0, // Gatsby onPostBuild life cycle
+      benchmarkEnd: 0, // End of benchmark itself
+    }
+    this.started = false
+  }
+
+  getData() {
+    for (const key in this.timestamps) {
+      this.timestamps[key] = Math.floor(this.timestamps[key])
+    }
+
+    const pageCount = parseInt(execSync("find . | grep page-data | wc -l", {
```

Comment:
This will fail on ~Windows (unless someone have busybox or something like that) - should it be clarified somewhere or potentially handle the potential error here gracefully?

Reply:
I will clarify it in the docs.

I think it's okay (for now at least) to skip windows support. This is something we're sharing but that's intended to be used internally. PR's are welcome to fix / improve it.

Reply:
This works pretty well:
```
import { sync as glob } from "fast-glob"
const pageCount = glob('**/**', {
    cwd: './page-data',
    nocase: true,
  }).length
```

Reply:
Sure that works for me too :) Thanks!